### ISSUE TITLE
Record information about the resolved global.json usage in the SdkResolver 

### ISSUE THREAD

**Issue by baronfel** [dotnet/sdk#30711]
### Is your feature request related to a problem? Please describe.

The SDK Resolver resolves an SDK according to any global.json rules that the .NET Host finds, and the Host reports back information like the path to the global.json, the version requested by the global.json, etc.

We should ensure that the SDK resolver logs this information via MSBuild properties to aid debugging - internal MSBuild users have been stumped by why a particular SDK was chosen when global.jsons far above the project root were at fault.

### Describe the solution you'd like

We should log properties (or a single item with metadata) that provide the following data from the SDK Resolver: 

* the path of any resolved global.json file
* the version requested in the global.json

This would make it easier for people debugging a build to determine if a global.json is impacting the build unexpectedly.

**Comment by xoofx** [comment]
Hey, I'm trying to find a documentation about how the SDK global.json is resolved and can only find the following code:

https://github.com/dotnet/sdk/blob/517aa20b62a78c6556147685bf834a66465f76ab/src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs#L215-L234

Do I understand correctly that if we pass a `csproj` or `sln` to `dotnet build`, it's going to locate the `global.json` from the `csproj`/`sln` folder, otherwise it is going to use `Environment.CurrentDirectory`

**Comment by baronfel** [comment]
@xoofx yes, that's correct. The comment notes that the CurrentDirectory backstop is for evaluating in-memory projects only.

**Comment by baronfel** [comment]
in https://github.com/dotnet/msbuild/pull/9621 the MSBuild team added logging of all resolution results, so that covers quite a bit. The only part missing from this is the _requested_ version, if any, from the global.json.

**Comment by baronfel** [comment]
This is a good candidate for domestic cat to pick up.
